src/ntlm/mod.rs

```diff
@@ -398,10 +416,20 @@ impl Sspi for Ntlm {
             self.complete_auth_token(&mut [])?;
         }
 
-        SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token)?; // check if exists
+        let mut encrypted = if let Ok(buffer) = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Token) {
+            buffer
+        } else {
+            SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Stream)?
+        }
+        .buffer
+        .clone();
         let data = SecurityBuffer::find_buffer_mut(message, SecurityBufferType::Data)?;
+        encrypted.extend_from_slice(&data.buffer);
 
-        *data.buffer.as_mut() = self.recv_sealing_key.as_mut().unwrap().process(data.buffer.as_slice());
+        let signature = encrypted[0..16].to_vec();
+        let enc = encrypted[16..].to_vec();
```

Comment:
Question: do you know if the "invariant" `buffer.len() > 16` is checked somewhere? Inside `SecurityBuffer` perhaps? Just being cautious, because in Rust, when the index is out of bounds, such indexing, instead of corrupting the memory, will stop the program by panicking (similar to C++ stack unwinding when an exception is raised). However, panicking (or unwinding) across an FFI boundary is undefined behavior. (There is [a working group on that matter](https://blog.rust-lang.org/inside-rust/2020/02/27/ffi-unwind-design-meeting.html), but as of today it’s undefined.) FYI, the safe route is to use `.get`:
```rust
let signature = encrypted.get(0..16)?;
let enc = &encrypted[16..]; // the second one is fine, because of the above
```
You may have do something like `get(0..16).or_else(|| …)` in order to convert the `Option` into a `Result`.
Alternatively, do the usual business with something like `if encrypted.len() < 16`.